### PR TITLE
feat: Refactor ConfirmPopupComponent to use Html (FM-315)

### PR DESCRIPTION
**Pull request by miguelccodev**
# Changes
- feat: created confirm popup w/ html buttons
- feat: integrated confirm popup in pause popup for en lang
- chore: added unit tests
- chore: added scss util classes
- chore: consolidated popup styles into base popup scss file.

# How to test
- start the game
- select any level
- click pause
- choose select level or restart level
  - should show confirm popup and play "are you sure" audio
    - choosing no will close the popups and resume the game
    - choosing yes will perform select level or restart level, depending on previous interaction

Ref: [FM-315](https://curiouslearning.atlassian.net/browse/FM-315)

[FM-315]: https://curiouslearning.atlassian.net/browse/FM-315?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by bernhardccodev — APPROVED** [review]
All good on my end. Approved!

**Review by woodsjohn-codev — APPROVED** [review]

File: src/scenes/gameplay-scene.ts [hunk 1]
```diff
@@ -175,7 +175,7 @@ export class GameplayScene {
 
     this.pausePopupComponent.onClose((event) => {
       const { data } = event;
-
+      console.log(event);
```
Comment (ashwinnair-chimple) [inline src/scenes/gameplay-scene.ts:178]: console need to be removed

File: src/components/popups/pause-popup/pause-popup-component.ts [hunk 2]
```diff
@@ -11,21 +14,51 @@ export const PAUSE_POPUP_EVENT_DATA = {
...
-    this.selectLevelButton = new MapButton({ targetId });
+    this.selectLevelButton = new MapButton({ id: 'map-button', className: 'me-4', targetId });
     this.selectLevelButton.onClick(() => this.handleClick('select-level'));
     this.restartLevelButton = new RetryButtonHtml({ targetId });
     this.restartLevelButton.onClick(() => this.handleClick('restart-level'));
+    this.confirmPopup = new ConfirmPopupComponent({
+      hideClose: true
+    });
+    
+    this.audioPlayer = new AudioPlayer();
```
Comment (ashwinnair-chimple) [inline src/components/popups/pause-popup/pause-popup-component.ts:33]: new AudioPlayer(); instantiated twice.

File: src/components/popups/pause-popup/pause-popup-component.ts [hunk 3]
```diff
@@ -11,21 +14,51 @@ export const PAUSE_POPUP_EVENT_DATA = {
...
+  }
+
+  handleClick(data: any) {
+    if (lang === 'english') {
+      const unsub = this.confirmPopup.onClose((confirmData) => {
+        unsub();
+        this.handleConfirmClose(confirmData, data);
+      });
+
+      if (this.clickTimeout) clearTimeout(this.clickTimeout);
```
Comment (ashwinnair-chimple) [inline src/components/popups/pause-popup/pause-popup-component.ts:43]: Can we consider Clear Timeout and Set timeout in a seperate method for reusability and readability?

File: src/components/popups/pause-popup/pause-popup-component.ts [hunk 4]
```diff
@@ -11,21 +14,51 @@ export const PAUSE_POPUP_EVENT_DATA = {
...
+      
+    } else {
+      super.handleClick(data);
+    }
+  }
+
+  handleConfirmClose(confirmClickEvent: PopupClickEvent, pauseData: any) {
+    if (confirmClickEvent.data) super.handleClick(pauseData);
+
+    super.handleClick(false, 0);
```
Comment (ashwinnair-chimple) [inline src/components/popups/pause-popup/pause-popup-component.ts:57]: Can we add comment to this line to understand its specific purpose.

**Review by ashwinnair-chimple — COMMENTED** [review]
Added few comments

File: src/components/popups/pause-popup/pause-popup-component.ts (hunk 3 above)
Comment (miguelccodev) [reply src/components/popups/pause-popup/pause-popup-component.ts:43]: added `lodash-es` using `debuonce` function for this instead, for reusability.

**Review by ashwinnair-chimple — APPROVED** [review]
Approved

**Pull request merged**